### PR TITLE
1786 Adding real-to-real conversion builtin

### PR DESCRIPTION
**Pull request by oakleybrunt**
Closes part of #1786

### CODE REVIEW

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Nice work! I have some requests on testing and documentation, and there are some style issues to fix (see inline).

Tests with compilation pass, the examples build and the user guide builds and displays nicely. The changes are covered. The branch also needs to be updated with the latest master.

File: doc/user_guide/dynamo0p3.rst [hunk 1]
```diff
@@ -2681,10 +2681,11 @@ scheme presented below. Any new Built-in needs to comply with these rules.
       data-type conversion Built-ins, see rule 7 below).
 
 7) As in the case of Built-in field argument rules, the names of the
-   field data-type conversion Built-ins, ``int_X`` (converts field data
-   from ``real`` to ``integer``) and ``real_X`` (converts field data
-   from ``integer`` to ``real``), are the only exceptions for the
-   naming of Built-ins in Fortran above.
+   field data-type conversion Built-ins,
```
[the remark this answers is not in the document]
Comment (oakleybrunt) [reply doc/user_guide/dynamo0p3.rst:2684]: I don't think that any of the conversion builtins break these rules (5.4, 6.5) anymore. All start with the type that they operate on now: `int_to_real_X` operates on integer-valued fields - was called `real_X` (not prefixed with matching datatype)

File: src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+# -----------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py:1]: `pycodestyle` reports `src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py:119:1: W293 blank line contains whitespace`.

File: src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    # Test '__str__' method
+    first_invoke = psy.invokes.invoke_list[0]
+    kern = first_invoke.schedule.children[0].loop_body[0]
+    assert str(kern) == ("Built-in: real_to_real_X (convert a real-valued to a "
+                         "real-valued field)")
+
+    # Test code generation
+    code = str(psy.gen)
+
+    print(code)
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py:91]: The `print` statement is still there 😃

**Review by TeranIvy — CHANGES_REQUESTED** [review]
Almost there. There are some minor style and documentation changes to do. 

Please also update the branch with the latest master if it changes before the next review round.

File: src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    # Test '__str__' method
+    first_invoke = psy.invokes.invoke_list[0]
+    kern = first_invoke.schedule.children[0].loop_body[0]
+    assert str(kern) == ("Built-in: real_to_real_X (convert a real-valued to "
+                         "a real-valued field)")
+
+    # Test code generation
+    code = str(psy.gen)
+
+    print(code)
```
[the remark this answers is not in the document]
Comment (oakleybrunt) [reply src/psyclone/tests/domain/lfric/builtins/real_to_real_x_test.py:91]: Whoops! Can't believe I made the same mistake twice!

File: doc/user_guide/dynamo0p3.rst [hunk 5]
```diff
@@ -3265,28 +3256,53 @@ the same field (``X = min(a, X)``)::
...
-where **ifield2** is an ``integer_field_type`` of ``i_def`` precision
-and a ``real``-valued field *field1* can be of any :ref:`supported
-precisions <lfric-mixed-precision>` for ``GH_REAL`` fields (e.g.
-``r_tran`` for ``r_tran_field_type``).
+where **ifield2** is the only currently supported ``integer_field_type``
+(``i_def`` precision) and a ``real``-valued field *field1* can be of
+any :ref:`supported precisions <lfric-mixed-precision>` for ``GH_REAL``
+fields (e.g. ``r_tran`` for ``r_tran_field_type``).
+
+Built-ins which take a ``real`` field element for conversion to
```
[the remark this answers is not in the document]
Comment (oakleybrunt) [reply doc/user_guide/dynamo0p3.rst:3281]: After moving this block and generating the pdf, it stll feels a little lost, as though it is part of the real_to_real_X builtin section. Would it be clearer if this block was made into a note?

[the remark this answers is not in the document]
Comment (TeranIvy) [reply doc/user_guide/dynamo0p3.rst]: I think you're right for this one. I rephrased it and moved just below the conversion subtitle. How does this sound?
```
Conversion of ``real`` field elements
#####################################

Built-ins which take a ``real`` field for conversion to a field of
a different datatype or precision are denoted by the datatype that the
input ``real`` field will be converted to. A Built-in that converts a
``real`` to an ``integer`` field is denoted by the phrase **to_int**.
Likewise, a Built-in that converts a ``real`` to a ``real`` field is
denoted by the phrase **to_real**.
```

[the remark this answers is not in the document]
Comment (TeranIvy) [reply doc/user_guide/dynamo0p3.rst]: Rephrased this and pushed. I wanted to merge, but there is question of authorship in `doc/user_guide/conf.py` that needs to be discussed with PSyclons so I'll park this until the Monday telco.

**Review by TeranIvy — COMMENTED** [review]
All well now. Updated changelog and user guide for merge to master. If CI and Integration tests pass, this can go on.

**Pull request merged**